libraries/ESP8266WiFi/src/WiFiClientSecureBearSSL.cpp

```diff
@@ -0,0 +1,1298 @@
+/*
+  WiFiClientBearSSL- SSL client/server for esp8266 using BearSSL libraries
+  - Mostly compatible with Arduino WiFi shield library and standard
+    WiFiClient/ServerSecure (except for certificate handling).
+
+  Copyright (c) 2018 Earle F. Philhower, III
+
+  This library is free software; you can redistribute it and/or
+  modify it under the terms of the GNU Lesser General Public
+  License as published by the Free Software Foundation; either
+  version 2.1 of the License, or (at your option) any later version.
+
+  This library is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public
+  License along with this library; if not, write to the Free Software
+  Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+*/
+
+#define LWIP_INTERNAL
+
+#include <list>
+#include <errno.h>
+#include <algorithm>
+
+extern "C" {
+#include "osapi.h"
+#include "ets_sys.h"
+}
+#include "debug.h"
+#include "ESP8266WiFi.h"
+#include "WiFiClient.h"
+#include "WiFiClientSecureBearSSL.h"
+#include "lwip/opt.h"
+#include "lwip/ip.h"
+#include "lwip/tcp.h"
+#include "lwip/inet.h"
+#include "lwip/netif.h"
+#include "include/ClientContext.h"
+#include "c_types.h"
+
+namespace BearSSL {
+
+// BearSSL needs a very large stack, larger than the entire ESP8266 Arduino
+// default one.  This shared_pointer is allocated on first use and cleared
+// on last cleanup, with only one stack no matter how many SSL objects.
+std::shared_ptr<uint8_t> WiFiClientSecure::_bearssl_stack = nullptr;
+
+
+
+void WiFiClientSecure::_clear() {
+  // TLS handshake may take more than the 5 second default timeout
+  _timeout = 15000;
+
+  _sc = nullptr;
+  _sc_svr = nullptr;
+  _eng = nullptr;
+  _x509_minimal = nullptr;
+  _x509_insecure = nullptr;
+  _x509_knownkey = nullptr;
+  _iobuf_in = nullptr;
+  _iobuf_out = nullptr;
+  _now = 0; // You can override or ensure time() is correct w/configTime
+  _ta = nullptr;
+  setBufferSizes(16384, 512); // Minimum safe
+  _handshake_done = false;
+  _recvapp_buf = nullptr;
+  _recvapp_len = 0;
+  _oom_err = false;
+  _deleteChainKeyTA = false;
+}
+
+void WiFiClientSecure::_clearAuthenticationSettings() {
+  _use_insecure = false;
+  _use_fingerprint = false;
+  _use_self_signed = false;
+  _knownkey = nullptr;
+  _sk = nullptr;
+}
+
+
+WiFiClientSecure::WiFiClientSecure() : WiFiClient() {
+  _clear();
+  _clearAuthenticationSettings();
+  _certStore = nullptr; // Don't want to remove cert store on a clear, should be long lived
+  if (!_bearssl_stack) {
+    #if 1
```

Comment:
How does this tie in with #4622 ? Will it be removed once that PR is merged? If so, please add a reminder comment to the PR.

Reply:
Deleting it completely as that extra space really should be managed by the core (if at all), not a library.